### ISSUE TITLE
Cesium ion icon missing, or handling file:// base URL

### ISSUE THREAD

**Issue by vividos** [CesiumGS/cesium#8665]
I'm hosting CesiumJS in an Android WebView where the base URL is file:///android_asset/. In the view, the Cesium ion icon is not shown, due to an error processing the base URL.

Sandcastle example:
https://sandcastle.cesium.com/index.html#c=ZZBda8IwFIb/SuhNK2wJu3VV5ooXgjKweFeQ2By7sDSn5KPixv77klo3t+UqH+/znHPSc0OOUsHOSDIjadxOGWNcC4NS7Lm14FjLO1aAlb5l6WOlK32SWuCJFstytdvsnxflcr/broNgVA2hGrVFBVRhk11oevBSiQ0KH1MqSxdRb9mq5Q1YJlHf1waEdLTTTTqZDJo+dGjAojc1hAoaTmS0bcfb7IN4o6bfc3xG8Lb6Fach9iP1w8g3vsBmo+OvIWSpw9IZqZtsaCy5S3LrzgrmlSZhPcm2Q+NiJxmlzEHbKe7CVAdfv4GjtbURy9kVyoXsiRSzKqmH8gVqx6UGUyWkVuHjw8vRK1XKd6iSec5C/hemkIvQzksPRvFzjLw+zNeXS0ppzsLxP+UQ1YGbG+MX

The file:/// URL is changed to file:/, and the icon can't be accessed anymore. Interestingly, all else works without problems.

Browser: Chrome on Android, Firefox on Windows

**Comment by vividos** [comment]
I think it's because of this line:
https://github.com/CesiumGS/cesium/blob/master/Source/ThirdParty/Uri.js#L263
The authority part of the URL is empty in my case.

**Comment by OmarShehata** [comment]
I can reproduce this issue running CesiumJS on Android with Cordova. I think the authority being empty may be the issue. @vividos I'd be happy to review a pull request if you have a chance to put together a fix.

**Comment by vividos** [comment]
I can do a PR but I'm not skilled enough in JavaScript development to test the change. And it might interfere with other cases of URL handling...

**Comment by vividos** [comment]
Here's what I found so far:
- The `Uri.toString()` method formats the base URL of my WebView incorrectly; `file:///android_asset/...` becomes `file:/android_asset/...`, which prevents showing the image. This is fixed when PR #8669 is applied
- There is one more issue: The default Ion credit image tag that is created in https://github.com/CesiumGS/cesium/blob/cbc09093d5f27b701925ec06766d8b2eceef6f73/Source/Scene/CreditDisplay.js#L490 is sanitized using DOMpurify.sanitize() in the Credit.element getter: https://github.com/CesiumGS/cesium/blob/master/Source/Core/Credit.js#L93 and the `<img src="">` attribute is removed because it doesn't contain a `http://` or `https://` link, but a `file:// link`, which gets scrubbed by DOMpurify.
- I tried to fix the second issue by providing a default credit object in my JavaScript code:
```
Cesium.CreditDisplay.cesiumCredit =
    new Cesium.Credit('<a href="https://cesium.com/" target="_blank"><img src="Cesium/Assets/Images/ion-credit.png" title="Cesium ion"/></a>', true);
```
(note how the src has a relative link).
- This fix initially works correct and shows the Ion credit icon, but only until the Ion terrain provider is initialized. This is due to the code in `CreditDisplay.getDefaultCredit()`, which doesn't recognize that I already set a custom `cesiumCredit` object:
https://github.com/CesiumGS/cesium/blob/cbc09093d5f27b701925ec06766d8b2eceef6f73/Source/Scene/CreditDisplay.js#L488
(my guess is that the code should not check the local var `defaultCredit` but `CreditDisplay._cesiumCredit` instead.

So that's the current state of this issue; PR #8669 would help, but not completely fix the issue.
